doc/design/manifest.md

```diff
@@ -0,0 +1,589 @@
+                },
+                "scheme": "rsassa-pss-sha256",
+            },
+        },
+        "threshold": 1,
+    },
+    "snapshot": {...},
+    "timestamp": {...},
+},
+```
+
+### n.index.json
+
+```
+"_type": "index"
+```
+
+Lists owners and components. File is versioned and immutable, n is the version.
+
+Replaces the targets role in TUF.
+
+Example:
+
+```json
+"owners": {
+    "pingcap": {
+        "name": "PingCAP",
+        "keys": {
+            "4e777de0d275f9d28588dd9a1606cc748e548f9e22b6795b7cb3f63f98035fcb": {
+                "keyid_hash_algorithms": [
+                    "sha256",
+                    "sha512",
+                ],
+                "keytype": "rsa",
+                "keyval": {
+                    "public": "-----BEGIN PUBLIC KEY-----\nMIIBojANBgkqhkiG9w0BAQEFAAOCAY8AMIIBigKCAYEA0GjPoVrjS9eCqzoQ8VRe\nPkC0cI6ktiEgqPfHESFzyxyjC490Cuy19nuxPcJuZfN64MC48oOkR+W2mq4pM51i\nxmdG5xjvNOBRkJ5wUCc8fDCltMUTBlqt9y5eLsf/4/EoBU+zC4SW1iPU++mCsity\nfQQ7U6LOn3EYCyrkH51hZ/dvKC4o9TPYMVxNecJ3CL1q02Q145JlyjBTuM3Xdqsa\nndTHoXSRPmmzgB/1dL/c4QjMnCowrKW06mFLq9RAYGIaJWfM/0CbrOJpVDkATmEc\nMdpGJYDfW/sRQvRdlHNPo24ZW7vkQUCqdRxvnTWkK5U81y7RtjLt1yskbWXBIbOV\nz94GXsgyzANyCT9qRjHXDDz2mkLq+9I2iKtEqaEePcWRu3H6RLahpM/TxFzw684Y\nR47weXdDecPNxWyiWiyMGStRFP4Cg9trcwAGnEm1w8R2ggmWphznCd5dXGhPNjfA\na82yNFY8ubnOUVJOf0nXGg3Edw9iY3xyjJb2+nrsk5f3AgMBAAE=\n-----END PUBLIC KEY-----",
+                },
+                "scheme": "rsassa-pss-sha256",
+            },
+            "59a4df8af818e9ed7abe0764c0b47b4240952aa0d179b5b78346c470ac30278d": {
+                "keyid_hash_algorithms": [
+                    "sha256",
+                    "sha512",
+                ],
+                "keytype": "ed25519",
+                "keyval": {
+                    "public": "edcd0a32a07dce33f7c7873aaffbff36d20ea30787574ead335eefd337e4dacd",
+                },
+                "scheme": "ed25519",
+            },
+        },
+    },
+},
+"components": {
+    "tidb": {
+        "name": "Name of the Component",
+        "yanked": false,
+        "owner": "pingcap",
+        "url": "/name_of.json",
+        "length": 500,
+        "threshold": 1,
+    },
+},
+"default components": [
+    "tidb",
+    "tikv",
+    "pd",
+]
+```
+
+Each owner id and component id must be unique (TiUp should treat owner and component ids as distinct types, but ids must be unique within the union of the types). 
+
+### n.xxx.json
+
+```
+"_type": "component"
+```
+
+Where xxx is the id of the component, n is the version of the manifest, file is immutable.
+
+```
+"name": "Name of the Component",
```

Comment:
Prefer
```
"name": "Name of the Component",
"description": "This is a component",
"versions": {
    "0.0.1": {
        "amd64-darwin": {
            "yanked": false,
            "url": "/name-of-amd64-darwin-v0.0.1.tar.gz",
            "hashes": {
                "sha256": "141f740f53781d1ca54b8a50af22cbf74e44c21a998fa2a8a05aaac2c002886b",
                "sha512": "ef5beafa16041bcdd2937140afebd485296cd54f7348ecd5a4d035c09759608de467a7ac0eb58753d0242df873c305e8bffad2454aa48f44480f15efae1cacd0"
            },
            "length": 1001000499,
            "dependencies": {
                "foo": "1.0.0",
            },
        },
        "amd64-linux": {
            "yanked": false,
            "url": "/name-of/name-of_x86_64-apple-darwin_0.0.1.tar.gz",
            "hashes": {
                "sha256": "141f740f53781d1ca54b8a50af22cbf74e44c21a998fa2a8a05aaac2c002886b",
                "sha512": "ef5beafa16041bcdd2937140afebd485296cd54f7348ecd5a4d035c09759608de467a7ac0eb58753d0242df873c305e8bffad2454aa48f44480f15efae1cacd0"
            },
            "length": 1001000499,
            "dependencies": {
                "foo": "1.0.0",
            },
        }
    },
    "0.0.2":{ ... },
}
```

The reason why I prefer:
1. The CDN which we used doesn't support sub-directory and the additional advantage is we can keep the filename backward compatible (maybe we should support the current solution in short-terms).
2. Grouping distributed by version instead of platforms.

Reply:
> The CDN which we used doesn't support sub-directory and the additional advantage is we can keep the filename backward compatible

That seems fine, using subdirectories is just to make it easier to look at, it is not necessary and can be changed by changing the URLs pretty easily.

> Grouping distributed by version instead of platforms.

The reason for platform first is that it makes things marginally easier to search since the platform is fixed - the client is only interested in one platform at a time - but the client must search the versions, e.g., find the most recent, so it is nice to have them together. Of course since there are unlikely to be thousands of versions, it is easy enough to convert in software.

Why do you prefer versions first?